gnovm/pkg/doctest/README.md

```diff
@@ -0,0 +1,131 @@
+# Gno Doctest: Easy Code Execution and Testing
+
+Gno Doctest is a tool that allows you to easily execute and test code blocks written in the Gno language. This tool offers a range of features, from simple code execution to complex package imports.
+
+## Basic Usage
+
+To use Gno Doctest, run the following command:
+
+gno doctest -path <markdown_file_path> -run <code_block_name | "">
+
+- `<markdown_file_path>`: Path to the markdown file containing Gno code blocks
+- `<code_block_name>`: Name of the code block to run (optional)
+
+For example, to run the code block named "print hello world" in the file "foo.md", use the following command:
+
+gno doctest -path foo.md -run "print hello world"
+
+## Features
+
+### 1. Basic Code Execution
+
+Gno Doctest can execute simple code blocks:
+
+```go
+package main
+
+func main() {
+    println("Hello, World!")
+}
+
+// Output:
+// Hello, World!
+```
+
+Doctest also recognizes that a block of code is a gno. The code below outputs the same result as the example above.
+
+```go
+// @test: print hello world
+package main
+
+func main() {
+    println("Hello, World!")
```

Comment:
the println does not print correct results compare with file tests.


  package main
  type ints []int
  func main() {
        a := ints{1,2,3}
      println(a)
  }

doctest

// Output:
// (slice[(1 int),(2 int),(3 int)] gno.land/r/g14ch5q26mhx3jk5cxl88t278nper264ces4m8nt/run.ints)

file test result is correct ints is a type defined in main package. 
 // Output:
 // (slice[(1 int),(2 int),(3 int)] main.ints)

Reply:
I think it print out the original caller's address instead of the main. It seems to be related to the implementation of `ExecutionCodeBlock` in `exec.go`. I'll need to look at it. thanks!

Reply:
Maybe we can replace it with the proper type when displaying it in the Markdown results.